### PR TITLE
fix: fix async validate type

### PR DESCRIPTION
**Pull request by yi-tuan**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export type ValidationValue = boolean | number | string | RegExp;
 
 export type ValidateResult = string | boolean | undefined;
 
-export type Validate = (data: any) => ValidateResult;
+export type Validate = (data: any) => ValidateResult | Promise<ValidateResult>;
```
Comment (bluebill1049) [inline src/types.ts]: thanks for your PR, why do you return promise?

Comment (bluebill1049) [reply src/types.ts]: even async function should return `string` or `boolean` right? (do i miss another use case)

Comment (yi-tuan) [reply src/types.ts]: image
async function should return `Promise`.

Comment (bluebill1049) [reply src/types.ts]: oh, I see. thanks! my bad.

Comment (yi-tuan) [reply src/types.ts]: This library is very useful, I should thank you.

Comment (bluebill1049) [reply src/types.ts]: my pleasure @yi-tuan thanks for your help too. ❤️

**Pull request merged**